### ISSUE TITLE
Feature Request: Please make Client remember the Avatar

### ISSUE THREAD

**Issue by 0lhi** [mumble-voip/mumble#4735]
I'd like to request to make the Client remember the Avatar and automatically upload to every server that it's connecting to.

Right now, users who use the Overlay have the choice between manually uploading the Avatar on every reconnect, or making an Account on every server they use, neither of which is particularly appealing.

**Comment by Krzmbrzl** [comment]
What Avatar? And how do you upload it? And where does it get displayed? :eyes:

**Comment by Kissaki** [comment]
Self -> Change Avatar

It’s bound to the user account, or connection if not registered.

**Comment by Krzmbrzl** [comment]
This Avatar is only used in the Overlay though, correct?

**Comment by Kissaki** [comment]
When you mouse hover over users you can see the avatar. When you click the info/comment icon you can see it too.

**Comment by gbidkar** [comment]
But what if I want a different avatar on different servers? Maybe it should be having a 'main avatar' which is the default one that will be displayed on each server like the original issue describes and a per-server avatar.

**Comment by colin-mcl** [comment]
Hi! I am looking to contribute my help to this project and thought that this issue looks around my skill level. I know C++ well but have never used Qt, is this a good place for me to start?

**Comment by Krzmbrzl** [comment]
@colin-mcl yes, I think this should be easily doable for you. My guess is that the biggest part of the required changes won't require Qt anyway and the UI side of this seems to be relatively straight forward as well - so probably a good start to get in touch with Qt :)

Do you want me to look up some relevant code parts that should get you started understanding the current logic of the code?

**Comment by colin-mcl** [comment]
That would be fantastic, thank you!

**Comment by Krzmbrzl** [comment]
Alright - I am unfamiliar with the precise workings of the avatar feature in Mumble (as you see above, I was unaware of this feature prior to this request) but I can pick out some parts, that I think are relevant nonetheless. Just be aware that there might be other parts that are important for this, that I don't list here.

- The current UI integration happens via the actions `qaServerTexture` (Self > Change Avatar) and `qaServerTextureRemove` (Self > Remove Avatar) in `MainWindow.cpp`
- Thus, the relevant code that is triggered by clicking any of these actions is
https://github.com/mumble-voip/mumble/blob/ba767c5e2da75c77444e5631b67b36730f6e5b3f/src/mumble/MainWindow.cpp#L1560-L1569
and
https://github.com/mumble-voip/mumble/blob/ba767c5e2da75c77444e5631b67b36730f6e5b3f/src/mumble/MainWindow.cpp#L1571-L1573
- As you see, both of these call
https://github.com/mumble-voip/mumble/blob/ba767c5e2da75c77444e5631b67b36730f6e5b3f/src/mumble/ServerHandler.cpp#L1029-L1084
- The server-side handling happens inside the `msgUserState` function at
https://github.com/mumble-voip/mumble/blob/ba767c5e2da75c77444e5631b67b36730f6e5b3f/src/murmur/Messages.cpp#L779-L795
and
https://github.com/mumble-voip/mumble/blob/ba767c5e2da75c77444e5631b67b36730f6e5b3f/src/murmur/Messages.cpp#L823-L837

Though, I think you don't need to make any server-side changes in order to implement this feature request :eyes:

**Comment by colin-mcl** [comment]
@Krzmbrzl This is great thank you so much. I think I understand the request flow that occurs when a User sets their avatar based on these code snippets and my own exploration. It seems that the simplest solution is to remember the user's default avatar once it is set by them. I'm a little unclear on how the UserState variables are recorded so I am going to dig through the code and try to get an understanding of that so that I could implement this solution. If I need any more help I will reach out to you!

**Comment by Krzmbrzl** [comment]
> It seems that the simplest solution is to remember the user's default avatar once it is set by them.

This would then have to be per-server though as we can't assume that a user wants to use the same avatar across different servers. 
In addition to that, I think we could add a setting to set up a default avatar, which will be used when connecting to a server we don't have a server-specific avatar stored yet.
